src/recursive_sorting/recursive_sorting.py

```diff
@@ -3,15 +3,46 @@ def merge( arrA, arrB ):
     elements = len( arrA ) + len( arrB )
     merged_arr = [0] * elements
     # TO-DO
+    new_arr = []
+    
+    for i in range(0, elements):
+        if len(arrA) == 0:
+            new_arr.append(arrB[0])
+            arrB.pop(0)
+            continue
+        if len(arrB) == 0:
+            new_arr.append(arrA[0])
+            arrA.pop(0)
+            continue
+        
+        if arrA[0] < arrB[0]:
+            new_arr.append(arrA[0])
+            arrA.pop(0)
+        else:
+            new_arr.append(arrB[0])
+            arrB.pop(0) 
+
+    return new_arr
+
```

Comment:
Nice job getting a working solution for this!

The next thing to consider here is how to do it without .append and .pop.  Later in the course you will learn why these methods are a bit more costly in operations than you might think.  This solution is totally fine for now, if you are interested in how to do it without .append and .pop, it will require keeping track of a couple extra index counters.  If you are interested in looking into this later, one approach is as below:

To do this 'in place', you would make use of the merged_arr defined on line 4.  This is just an array that is the final size but every value is initialized as 0.  You would need to create an index counter for each arrA and arrB to keep track of what has been inserted from each.  You could initialize both counters at 0 to look at the first elements of each.  In the for loop Compare the items at those indices, whichever is smaller put into the merged_arr at 'i' (as from the for loop) and increase the array it came from's index counter by 1. At some point one of the arrays will be used completely (its index counter will be larger than its arrays length) in which case you no longer need to compare and can just continue dumping the items from the other array into merged_arr until your for loop is complete.